ibazel/ibazel.go

```diff
@@ -340,8 +407,19 @@ func verb(s string) string {
 	}
 }
 
+func (i *IBazel) cancel() {
+	if i.b != nil {
+		i.b.Cancel()
+	}
+
+	if i.cmd != nil {
+		i.cmd.Kill()
+	}
+}
+
 func (i *IBazel) build(targets ...string) (*bytes.Buffer, error) {
```

Comment:
Break this out into 2 functions `buildWithEarlyAbort` and just `build` and then inline the logic here. I'd like to avoid storing `i.b` if at all possible. That's too much state to be managing.